### ISSUE TITLE
Log exception details

### ISSUE THREAD

**Issue by georgechr** [huorswords/Microsoft.Extensions.Logging.Log4Net.AspNetCore#1]
At the moment Log4Net for AspNetCore seems to ignore the Exception object when logging Errors or Warnings

**Comment by huorswords** [comment]
Hi @georgechr ,

Thank you very much for your feedback.

Can you provide some additional information in order to reproduce the error on my side?

Thank you. Regards,
--
Angel

**Comment by huorswords** [comment]
I've tried to reproduce the problem but I think that I didn't reproduced it correctly.

Steps
1. Create a Web API .NET Core project by default.
2. Configure the extensions on Startup.cs file properly.
3. Throw an exception from the API/Values default controller action.

The result was that the exception (and the stack trace) was printed on the debug window.

```
Lab> fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[0]
Lab>       An unhandled exception has occurred while executing the request
Lab> System.Exception: Doh!
Lab>    at Lab.Controllers.ValuesController.Get() in C:\angel\code\github\huorswords\Microsoft.Extensions.Logging.Log4Net.AspNetCore\src\Lab\Controllers\ValuesController.cs:line 16
Lab>    at lambda_method(Closure , Object , Object[] )
Lab>    at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionMethodAsync>d__12.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeNextActionFilterAsync>d__10.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeNextResourceFilter>d__22.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Lab>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Lab>    at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
Lab> --- End of stack trace from previous location where exception was thrown ---
Lab>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Lab>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Lab>    at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.<Invoke>d__7.MoveNext()
```

I can't see what you where or what the problem is, so @georgechr please, provide me some information (or an example project) with your expected result and steps to reproduce it.

Thanks.

Angel

**Comment by georgechr** [comment]
Hello,

I am adding the logger to the logger factory in a vanilla Core app (not an MVC one). I will post repro steps later on.

**Comment by huorswords** [comment]
Hello @georgechr,

Thank you very much by your report.

I just added a change on the develop branch in order to use the content of the exception as part of the final message that will be logged.

And, for extensibility questions, I've added also the possibility of configure your own formatter when an exception is included as part of the logged state.

Let me know if these changes will fix your issue.

Thanks
